.murdock

```diff
@@ -131,6 +136,22 @@ print_worker() {
         echo "-- running on worker ${DWQ_WORKER} thread ${DWQ_WORKER_THREAD}, build number $DWQ_WORKER_BUILDNUM."
 }
 
+test_hash_calc() {
+    local appdir=$1
+    local bindir=$2
+
+    md5sum $appdir/tests/* $bindir/*.elf | md5sum | cut -f1 -d' '
+}
+
+test_cache_get() {
+    test "${ENABLE_TEST_CACHE}" = "1" || return 1
+    test -n "$(redis-cli get $1)" > /dev/null
```

Comment:
Any reference to the documentation of the added requirement?

Reply:
No. Agreed, some lines somewhere describing the mechanism are needed.

But this is not a hard requirement. Apart from a couble of "redis-cli not found" error messages, nothing of this breaks a build. The cache just won't work if anything is missing.

Reply:
> But this is not a hard requirement. Apart from a couble of "redis-cli not found" error messages, nothing of this breaks a build. The cache just won't work if anything is missing.

I don't like having publicly running systems just run into error cases. Then rather have it documented somewhere and skipped if it isn't installed:

```suggestion
    test -n "$(command -v redis-cli > /dev/null && redis-cli get $1)" > /dev/null
```

Reply:
+1. Is the ">/dev/null" necessary for command? On my arch linux it just has empty output when a command is not found.

Reply:
Yepp, but it yields an output *if* the command is found ;-).

Reply:
Ah right! I read that as ```test -n $(command -v redis-cli) && test -n $(redis-cli get...)```